UnityProject/Assets/Scripts/Tilemaps/Behaviours/Objects/RegisterTile.cs

```diff
@@ -205,6 +205,22 @@ protected virtual void Awake()
 		IPlayerEntersTiles = GetComponents<IPlayerEntersTile>();
 		IObjectEntersTiles = GetComponents<IObjectEntersTile>();
 		CurrentsortingGroup = GetComponent<SortingGroup>();
+
+		if (GameManager.Is3D)
+		{
+			var Is3D = this.gameObject.GetComponent<ConvertTo3D>();
```

Comment:
the naming suggest this is a boolean 
```suggestion
			var convertTo3d = this.gameObject.GetComponent<ConvertTo3D>();
```